File: MapTest/results/MapTest.pl
use strict;
use warnings;

package TestClass;

sub new
{
    my $class = shift;
    my $self = {};
    bless $self, $class;
    return $self;
}

sub get_result {
    my ( $self ) = @_;
    my $map_obj = {
      x => 5,
    };
    #let containsX = "x" in mapObj;
    #delete mapObj["x"];
    ${$map_obj}{"x"} = 3;
    return ${$map_obj}{"x"};
}

sub test_method {
    my ( $self ) = @_;
    print(("Result = @{[$self->get_result()]}") . "\n");
}

package Program;

eval {
    my $c = new TestClass();
    $c->test_method();
};
if ($@) {
    print "Exception: " . $@
}

File: MapTest/results/MapTest.cs
using System.Collections.Generic;
using System.Linq;
using System;

public class TestClass
{
    public int GetResult()
    {
        var mapObj = new Dictionary<string, int>
        {
          { "x", 5 }
        };
        //let containsX = "x" in mapObj;
        //delete mapObj["x"];
        mapObj["x"] = 3;
        return mapObj["x"];
    }
    
    public void TestMethod()
    {
        Console.WriteLine($"Result = {this.GetResult()}");
    }
}

public class Program
{
    static public void Main(string[] args)
    {
        try 
        {
            new TestClass().TestMethod();
        }
        catch (System.Exception e)
        {
            System.Console.WriteLine($"Exception: {e.Message}");
        }
    }
}